src/ci/docker/host-x86_64/x86_64-gnu-llvm-12-stage1/Dockerfile

```diff
@@ -0,0 +1,53 @@
+FROM ubuntu:20.04
+
+ARG DEBIAN_FRONTEND=noninteractive
+RUN apt-get update && apt-get install -y --no-install-recommends \
+    g++ \
+    gcc-multilib \
+    make \
+    ninja-build \
+    file \
+    curl \
+    ca-certificates \
+    python2.7 \
+    git \
+    cmake \
+    sudo \
+    gdb \
+    llvm-12-tools \
+    llvm-12-dev \
+    libedit-dev \
+    libssl-dev \
+    pkg-config \
+    zlib1g-dev \
+    xz-utils \
+    nodejs
+
+COPY scripts/sccache.sh /scripts/
+RUN sh /scripts/sccache.sh
+
+# using llvm-link-shared due to libffi issues -- see #34486
+ENV RUST_CONFIGURE_ARGS \
+    --build=x86_64-unknown-linux-gnu \
+    --llvm-root=/usr/lib/llvm-12 \
+    --enable-llvm-link-shared \
+    --set rust.thin-lto-import-instr-limit=10
+
+ENV SCRIPT python2.7 ../x.py --stage 1 test --exclude src/tools/tidy && \
+    # Run the `mir-opt` tests again but this time for a 32-bit target.
+    # This enforces that tests using `// EMIT_MIR_FOR_EACH_BIT_WIDTH` have
+    # both 32-bit and 64-bit outputs updated by the PR author, before
+    # the PR is approved and tested for merging.
+    # It will also detect tests lacking `// EMIT_MIR_FOR_EACH_BIT_WIDTH`,
+    # despite having different output on 32-bit vs 64-bit targets.
+    python2.7 ../x.py --stage 1 test src/test/mir-opt \
+    --host='' --target=i686-unknown-linux-gnu && \
```

Comment:
I suspect this will fail unless you add a bunch of excludes, see the list in https://github.com/rust-lang/rust/pull/99141/files#diff-5af50d95255aad85107312975f70a64e701c1eeee029cc933205297fd3863992R38-R40
but maybe those are ok with --stage 1 and only break with --stage 0? not sure, fine to leave as is unless CI fails

Reply:
yeah, will see if ci fails.

Reply:
ci isn't failing, will resolve